src/merger/merger.c

```diff
@@ -494,6 +500,15 @@ luaL_merge_source_buffer_destroy(struct merge_source *base)
 	free(source);
 }
 
+static inline size_t
+ibuf_used(char *rpos, char *wpos)
+{
+	if (!rpos && !wpos)
+		return 0;
```

Comment:
1. Rule of thumb: explicitly check against NULL.
2. I would not just return 0 for an invalid buffer. Assertion fail is better than some subtle output data corruption.

Reply:
1. Done
2. Partly disagreed. `assert()` is only working for debug builds, but for release omitting sanity checks and blindly using incorrect data is worst approach for production. Collecting coredump is easy for debugging, but worse 100x times that check of input arguments and bailing out accordingly. 
In prior life (while being at Acronis and Ya.Taxi we always preferred both - `assert` and `if ... return`. Bceuase they are working at different times.

Reply:
Let's set/return an error and handle it in all usages so.